src/metabase/util/honeysql_extensions.clj

```diff
@@ -83,8 +83,11 @@
        (for [component components]
          (hformat/quote-identifier component, :split false)))))
   pretty/PrettyPrintable
-  (pretty [_]
-    (cons `identifier (cons identifier-type components))))
+  (pretty [this]
+    (if (= (set (keys this)) #{:identifier-type :components})
+      (cons `identifier (cons identifier-type components))
+      ;; if there's extra info beyond the usual two keys print with the record type reader literal syntax e.g. #metabase..Identifier {...}
+      (list (symbol (str \# `Identifier)) (into {} this)))))
```

Comment:
I tweaked the pretty-print method for `hx/Identifier` to print it with normal record type reader literal syntax e.g.

```clojure
#metabase.util.honeysql_extensions.Identifier {...}
```

if it contains extra keys beyond the usual `identifier-type` and `components` keys. This is so it will print any additional info that a driver might add, like the `::do-not-qualify?` info that we add